### PR TITLE
Add an early error when trying to send tokens to self

### PR DESCRIPTION
**Pull request by MuncleUscles**

### CODE REVIEW

File: test/integration/TokenTransfers.spec.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+} from '../../src'
+
+const ERROR_MESSAGE = 'Local node needs to be running to run these tests'
+
+describe('Token transfers', () => {
+    const identityManager = new RadixIdentityManager()
+
+    let identity1: RadixIdentity
+    let account2: RadixAccount
+    let TBD_URI: string
```
Comment (laszlourszuly) [inline test/integration/TokenTransfers.spec.ts:22]: Nitpicking, but technically this isn't a `URI`, but rather an `RRI`.
(URI = `scheme:[//authority]path[?query][#fragment]`)

**Review by laszlourszuly — COMMENTED** [review]
Looks great to me! Nice test-suite you've added.

File: test/integration/TokenTransfers.spec.ts (hunk 1 above)
Comment (MuncleUscles) [reply test/integration/TokenTransfers.spec.ts:22]: Hmm, it seems to me that a URI is a more broad category that applies to any Unique Resource Identifiers, what you have specified is a URL scheme https://tools.ietf.org/html/rfc3986#section-1.1.2

Comment (laszlourszuly) [reply test/integration/TokenTransfers.spec.ts:22]: Yes, that is correct; URL and URN are subsets of URI, but that was not my point.

My point was that a `URI` requires a `scheme` and `TBD_URI` never defines a `scheme` - we don't have a `radix:` scheme, or any other scheme for that matter, for any of our resources, hence, technically speaking, they are not `URI`'s.

But again, this is all nitpicking and nothing that should be taken too seriously.

**Pull request merged**